internal/controller/cluster_upgrade.go

```diff
@@ -697,39 +697,50 @@ func (r *ClusterReconciler) upgradeInstanceManager(
 		}
 		operatorHash := targetManager.GetHash()
 
-		if instanceManagerHash != "" && instanceManagerHash != operatorHash && !instanceManagerIsUpgrading {
-			// We need to upgrade this Pod
-			contextLogger.Info("Upgrading instance manager",
-				"pod", postgresqlStatus.Pod.Name,
-				"oldVersion", postgresqlStatus.ExecutableHash)
-
-			if cluster.Status.Phase != apiv1.PhaseOnlineUpgrading {
-				err := r.RegisterPhase(ctx, cluster, apiv1.PhaseOnlineUpgrading, "")
-				if err != nil {
-					return err
-				}
-			}
+		if instanceManagerIsUpgrading || instanceManagerHash == "" || instanceManagerHash == operatorHash {
+			message := fmt.Sprintf("Instance manager will skip upgrade on %s (upgrading: %t) "+
+				"(operator hash: %s — instance manager hash: %s)",
+				postgresqlStatus.Pod.Name,
+				instanceManagerIsUpgrading,
+				operatorHash[:6],
+				instanceManagerHash[:6])
+			contextLogger.Trace(message)
```

Comment:
@mnencia why this is a Trace and not Debug? the next message is Info level and it's completely in context with this one, so I think this is Debug or Info level but not Trace for sure, what do you think ?